strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java

```diff
@@ -133,109 +153,84 @@ public ResponseEntity create(@RequestBody @Validated(UserForm.NewUser.class) Use
-                                  @RequestHeader(HttpHeaders.ACCEPT) String accept)
-    {
-        User user = userService.findByUserName(name);
-        if (user == null)
-        {
-            return getNotFoundResponseEntity(NOT_FOUND_USER, accept);
-        }
-
-        UserOutput userOutput = UserOutput.fromUser(user);
-        Object body = getUserOutputEntityBody(userOutput, accept);
-        return ResponseEntity.ok(body);
-    }
-
-    @ApiOperation(value = "Used to retrieve all users")
-    @ApiResponses(value = { @ApiResponse(code = 200, message = SUCCESSFUL_GET_USERS) })
-    @PreAuthorize("hasAuthority('VIEW_USER')")
-    @GetMapping(value = "/all",
-            produces = { MediaType.APPLICATION_JSON_VALUE })
-    @ResponseBody
-    public ResponseEntity getUsers()
-    {
-        List<UserOutput> users = userService.findAll()
-                                            .getUsers()
-                                            .stream()
-                                            .map(UserOutput::fromUser).collect(Collectors.toList());
-
-        return getJSONListResponseEntityBody("users", users);
-    }
-
-    @ApiOperation(value = "Used to update user")
+    @ApiOperation(value = "Used to update an existing user")
     @ApiResponses(value = { @ApiResponse(code = 200, message = SUCCESSFUL_UPDATE_USER),
-                            @ApiResponse(code = 400, message = FAILED_UPDATE_USER) })
-    @PreAuthorize("hasAuthority('UPDATE_USER') || #userToUpdate.username == principal.username")
-    @PutMapping(value = "user",
-            consumes = MediaType.APPLICATION_JSON_VALUE,
-            produces = { MediaType.TEXT_PLAIN_VALUE,
-                         MediaType.APPLICATION_JSON_VALUE })
+                            @ApiResponse(code = 400, message = FAILED_UPDATE_USER),
+                            @ApiResponse(code = 403, message = USER_DELETE_FORBIDDEN) })
+    @PreAuthorize("hasAuthority('UPDATE_USER')")
+    @PutMapping(value = "{username}",
+                consumes = MediaType.APPLICATION_JSON_VALUE,
+                produces = { MediaType.TEXT_PLAIN_VALUE,
+                             MediaType.APPLICATION_JSON_VALUE })
     @ResponseBody
-    public ResponseEntity update(@RequestBody @Validated(UserForm.ExistingUser.class) UserForm userToUpdate,
+    public ResponseEntity update(@ApiParam(value = "The name of the user", required = true)
+                                 @PathVariable String username,
+                                 @RequestBody @Validated(UserForm.ExistingUser.class) UserForm userToUpdate,
                                  BindingResult bindingResult,
                                  Authentication authentication,
                                  @RequestHeader(HttpHeaders.ACCEPT) String accept)
     {
+        // If the {username} is admin - don't allow editing!
+        if (StringUtils.equals(username, "admin"))
```

Comment:
There is no guarantee that there will be exactly one admin user. In fact, if the XML config is not the configured authentication provider (let's say we're using LDAP), the configuration may be such that this user is no longer present, as the owner of the service may have decided to delete it and have the admins "imported" from LDAP.

Reply:
Please review SB-868, @carlspring